test/compile/data/bin.test.ts

```diff
@@ -12,53 +12,142 @@ describe('compile/data/bin', function() {
               'field': 'Acceleration',
-              'type': "quantitative"
+              'type': 'quantitative'
             }
           }
         });
         const transform = vals(bin.parseUnit(model))[0];
         assert.deepEqual(transform[0], {
           type: 'bin',
           field: 'Acceleration',
-          as: ['bin_Acceleration_start', 'bin_Acceleration_end'],
+          as: ['bin_extent_0,100_maxbins_10_Acceleration_start', 'bin_extent_0,100_maxbins_10_Acceleration_end'],
           maxbins: 10,
           extent: [0, 100],
-          signal: "Acceleration_bins",
+          signal: 'bin_extent_0_100_maxbins_10_Acceleration_bins',
         });
       });
     });
 
     describe('binned field without custom extent', () => {
       const model = parseUnitModel({
-        mark: "point",
+        mark: 'point',
         encoding: {
           y: {
             bin: true,
             'field': 'Acceleration',
-            'type': "quantitative"
+            'type': 'quantitative'
           }
         }
       });
       const transform = vals(bin.parseUnit(model))[0];
 
       it('should add bin transform and correctly apply bin', function() {
+        assert.deepEqual(transform.length, 2);
         assert.deepEqual(transform[0], {
           type: 'extent',
           field: 'Acceleration',
-          signal: 'Acceleration_extent'
+          signal: 'bin_maxbins_10_Acceleration_extent'
         });
         assert.deepEqual(transform[1], {
           type: 'bin',
           field: 'Acceleration',
-          as: ['bin_Acceleration_start', 'bin_Acceleration_end'],
+          as: ['bin_maxbins_10_Acceleration_start', 'bin_maxbins_10_Acceleration_end'],
           maxbins: 10,
-          signal: "Acceleration_bins",
-          extent: {signal: 'Acceleration_extent'}
+          signal: 'bin_maxbins_10_Acceleration_bins',
+          extent: {signal: 'bin_maxbins_10_Acceleration_extent'}
+        });
+      });
+    });
+    describe('same field encoded twice', () => {
+      const model = parseUnitModel({
+        data: {'url': 'data/movies.json'},
+        mark: 'circle',
+        encoding: {
+          x: {
+            bin: {maxbins: 10},
+            field: 'IMDB_Rating',
+            type: 'q'
+          },
+          y: {
+            bin: {'maxbins': 10},
+            field: 'Rotten_Tomatoes_Rating',
+            type: 'q'
+          },
+          color: {
+            bin: {maxbins: 20},
+            field: 'Rotten_Tomatoes_Rating',
+            type: 'quantitative'
+          },
+          size: {
+            aggregate: 'count',
+            field: '*',
+            type: 'q'
+          }
+        }
+      });
+      const transformArray = bin.parseUnit(model);
+      const transformArrayVals = vals(transformArray);
+      const imdbTransform = transformArrayVals[0];
+      const rotten10Transform = transformArrayVals[1];
+      const rotten20Transform = transformArrayVals[2];
+      it('should be binned twice', function() {
```

Comment:
This test and its description are a bit confusing.

1) The binned field that is encoded twice should be bin *once*, not twice.
2) The description is confusing partly because the test spec is a bit too complicated. Why do you need to include `x: IMDB_Rating` and `size: count`? I think better use just a minimal spec that demonstrates the thing you want to test.